crates/ide-diagnostics/src/handlers/type_mismatch.rs

```diff
@@ -18,24 +17,18 @@ pub(crate) fn type_mismatch(ctx: &DiagnosticsContext<'_>, d: &hir::TypeMismatch)
     let display_range = match &d.expr_or_pat {
         Either::Left(expr) => {
             adjusted_display_range::<ast::Expr>(ctx, expr.clone().map(|it| it.into()), &|expr| {
-                if !expr.is_block_like() {
-                    return None;
-                }
-
                 let salient_token_range = match expr {
                     ast::Expr::IfExpr(it) => it.if_token()?.text_range(),
                     ast::Expr::LoopExpr(it) => it.loop_token()?.text_range(),
                     ast::Expr::ForExpr(it) => it.for_token()?.text_range(),
                     ast::Expr::WhileExpr(it) => it.while_token()?.text_range(),
                     ast::Expr::BlockExpr(it) => it.stmt_list()?.r_curly_token()?.text_range(),
                     ast::Expr::MatchExpr(it) => it.match_token()?.text_range(),
-                    _ => {
-                        never!();
-                        return None;
-                    }
+                    ast::Expr::MethodCallExpr(it) => it.name_ref()?.ident_token()?.text_range(),
```

Comment:
we might as well add await and field expression as well? Since those would also underline the receiver otherwise

Reply:
good call!